File: chroma-league/src/main/java/net/booone/chromaleague/hud/animations/StaticBlinkingAnimation.java
package net.booone.chromaleague.hud.animations;

import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.SimpleFrame;
import net.booone.razersdk.color.Color;
import net.booone.razersdk.color.StaticColor;

public abstract class StaticBlinkingAnimation extends AnimatedFrame {
    private static final int BLINK_TIMES = 8;

    protected StaticBlinkingAnimation(Color color) {
        for (int i = 0; i < BLINK_TIMES; i++) {
            addAnimationFrame(new SimpleFrame(color));
            addAnimationFrame(3, new SimpleFrame(StaticColor.BLACK));
        }
    }
}

File: src/hud/animations/StaticBlinkingAnimation.ts
import {AnimatedFrame} from '../../razer-sdk/animation/AnimatedFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';
import {Color} from '../../razer-sdk/color/Color.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';

const BLINK_TIMES = 8;

export class StaticBlinkingAnimation extends AnimatedFrame {
    constructor(color: Color) {
        super();
        for (let i = 0; i < BLINK_TIMES; i++) {
            this.addAnimationFrame(new SimpleFrame(color));
            this.addAnimationFrame(3, new SimpleFrame(StaticColor.BLACK));
        }
    }
}
